src/azul/portal_service.py

```diff
@@ -0,0 +1,64 @@
+from copy import deepcopy
+from typing import (
+    Optional,
+    Sequence,
+    cast,
+)
+import json
+import logging
+
+import boto3
+from botocore.exceptions import ClientError
+
+from azul import config
+from azul.types import JSON
+
+log = logging.getLogger(__name__)
+
+
+class PortalService:
+
+    @staticmethod
```

Comment:
This should be mentioned in `CONTRIBUTING.rst`: we don't use `@staticmethod`, only `@classmethod` even if the first argument (`cls`) of such a method is not used by its body. Whether `cls` is used is often incidental and an implementation detail. We wouldn't want to repeatedly switch from `@staticmethod` to `@classmethod` and back if that implementation detail changes. We'll simply declare all methods that should be invoked through the class (as opposed to through an instance of that class) as `@classmethod` and call it a day. The same consideration goes for instance methods and `self`: some use it, some don't. The ones that don't shouldn't suddenly be considered static methods. The distinction between instance and class methods is driven by higher order concerns than the one about whether a method's body currently references `self` or not.

Could you add a section to `CONTRIBUTING.rst` in a separate commit in this PR? Feel free to use some of the above text for that. It's the best and most in-depth explanation of this rule I've given so far.

Reply:
Aside from that: these methods should be instance methods and `app.py` should instantiate this class before calling them. This is a pattern that we should follow for all services and controllers.

Reply:
Sounds good for the most part, but `PortalService` currently is not used anywhere in `app.py`. Should an instance be created there that is accessed for use by other classes, or is it okay to create the instance in the `azul.project.hca.Plugin` class where it's actually used?